composer.json

```diff
@@ -10,20 +10,29 @@
       "homepage": "http://www.auth0.com/"
     }
   ],
+  "repositories": [
+    {
+      "type": "git",
+      "url": "https://github.com/auth0/php-jwt.git"
+    }
+  ],
   "require": {
-    "php": "^7.1",
-    "guzzlehttp/guzzle": "~6.0 | ~7.0",
+    "php": "^7.3 | ^8.0",
     "ext-json": "*",
-    "lcobucci/jwt": "^3.3",
+    "ext-openssl": "*",
```

Comment:
Ideally yes, I agree that would be preferred here. The problem with specifying versions in this manner for PHP extensions is that it will block the SDK from installing in environments running in-development/pre-release versions of PHP.

For example, we could specify `"ext-json": ^1.7 | ^7.4 | ^8.0`, and our SDK would install for all GA versions of PHP we support. However, it would block installation in any forthcoming snapshots, betas or RCs, such as imminent development around 7.5 or 8.1 releases. You'd be surprised how many people run those. We can get around this by specifying development versions but this will lead to a lot of maintenance on our end.

Besides, it isn't really terribly important what version of these extensions a user has installed here, as long as they have one and it's enabled. These extensions will only build and work for appropriate PHP versions, so it's impossible to have an out of date extension on an up to date and supported PHP version.